okhttp/src/main/java/com/squareup/okhttp/Headers.java

```diff
@@ -250,7 +250,8 @@ public Builder add(String name, String value) {
      * Add a field with the specified value without any validation. Only
      * appropriate for headers from the remote peer or cache.
      */
-    Builder addLenient(String name, String value) {
+    // public for MockWebServer to allow simulation of bad server responses.
+    public Builder addLenient(String name, String value) {
```

Comment:
You can go through `Internal.addLenient` to avoid having to put this in the public API.